server/api-util/lineItems.js

```diff
@@ -0,0 +1,47 @@
+const { calculateQuantityFromDates, calculateTotalFromLineItems } = require('./lineItemHelpers');
+
+const unitType = 'line-item/night';
+const PROVIDER_COMMISSION_PERCENTAGE = -10;
+
+/** Returns collection of lineItems (max 50)
+ *
+ * Each line items has following fields:
+ * - `code`: string, mandatory, indentifies line item type (e.g. \"line-item/cleaning-fee\"), maximum length 64 characters.
+ * - `unitPrice`: money, mandatory
+ * - `lineTotal`: money
+ * - `quantity`: number
+ * - `percentage`: number (e.g. 15.5 for 15.5%)
+ * - `seats`: number
+ * - `units`: number
+ * - `includeFor`: array containing strings \"customer\" or \"provider\", default [\":customer\"  \":provider\" ]
+ *
+ * Line item must have either `quantity` or `percentage` or both `seats` and `units`.
+ *
+ * `includeFor` defines commissions. Customer commission is added by defining `includeFor` array `["customer"]` and provider commission by `["provider"]`.
+ *
+ * @param {Object} listing
+ * @param {Object} bookingData
+ * @returns {Array} lineItems
+ */
+exports.transactionLineItems = (listing, bookingData) => {
+  const unitPrice = listing.attributes.unitPrice;
```

Comment:
Now that I got to testing this, I notice that listings don't have a `unitPrice` attribute. It should be just `price`.

```suggestion
  const unitPrice = listing.attributes.price;
```